### PR TITLE
Feature/linux deploy

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Fixes Linux deployment

Changes proposed in this PR:
 * Use LinuxDeployQt for copying Qt dependencies
 * CI: Disable Python on Mac/Linux since we cannot deploy it (might want to do something differ

### CODE REVIEW

File: apps/inviwo/packaging/packaging.cmake [hunk 1]
```diff
@@ -114,6 +114,31 @@ elseif(APPLE)
...
+      "Name=${IVW_PACKAGE_SELECT_APP}"
+      "Exec=${IVW_PACKAGE_SELECT_APP}"
+      "Icon=${IVW_PACKAGE_SELECT_APP}"
+      "Categories=Science;"
+    )
+    install(FILES ${PROJECT_BINARY_DIR}/${IVW_PACKAGE_SELECT_APP}.desktop 
+            DESTINATION share/applications 
+            COMPONENT Application
+    )
+    install(FILES ${IVW_ROOT_DIR}/resources/inviwo/inviwo-logo-512px.png
```
Comment (martinfalk) [inline apps/inviwo/packaging/packaging.cmake:127]: Any idea whether SVG icons are also supported?

**Review by martinfalk — DISMISSED** [review]

File: apps/inviwo/packaging/packaging.cmake (hunk 1 above)
Comment (ResearchDaniel) [reply apps/inviwo/packaging/packaging.cmake:127]: I believe so, but the first goal is to make it work.
One can dive into it here:
https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html

File: .github/workflows/inviwo.yml [hunk 2]
```diff
@@ -77,8 +82,15 @@ jobs:
       with:
         repository: 'microsoft/vcpkg'
         path: 'vcpkg' # The cmake vcpkg present expects vcpkg to be next to the inviwo source folder
-    
+
+    - name: 'Bootstrap vcpkg'
+      shell: bash
+      run: |
+        ${{ matrix.bootstrap_vcpkg }} 
+        ./vcpkg/vcpkg integrate install
```
Comment (petersteneteg) [inline .github/workflows/inviwo.yml]: No we do not want to use that, we use manifests and toolchains so we can control the dependencies

File: .github/workflows/inviwo.yml [hunk 3]
```diff
@@ -77,8 +82,15 @@ jobs:
...
+
     # Cache built dependencies for faster subsequent builds
+    # ./vcpkg/bootstrap-vcpkg.bat must have been called before
```
Comment (petersteneteg) [inline .github/workflows/inviwo.yml:92]: Since when?

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: .github/workflows/inviwo.yml (hunk 3 above)
Comment (ResearchDaniel) [reply .github/workflows/inviwo.yml:92]: Not sure if this is the issue with the Mac/Linux not working. Thought I would give it a try

https://nicedoc.io/Microsoft/vcpkg/docs/users/binarycaching.md#github-packages

Comment (petersteneteg) [reply .github/workflows/inviwo.yml:92]: What is not working for Mac/linux? The caching?

Comment (ResearchDaniel) [reply .github/workflows/inviwo.yml:92]: Yes, exactly, it fails when trying to upload. But I think that I have fixed it now (see change for nuget setup). Unsure if the bootstrapping before is necessary.

for some reason it does not work for the visualneuro repo. Perhaps the repo  is used in the nuget setup somehow

**Comment by petersteneteg** [comment]
@ResearchDaniel status? should we close this? We should create an AppImage on linux If we want an installer... just taring the install folder will be very very error prone.

**Comment by ResearchDaniel** [comment]
@petersteneteg This PR tries to do that using linuxdeployqt. I have limited possibilities for testing this beyond github actions and it is not on my prio list. 

 Perhaps we can at least extract the nuget binary caching into a separate PR

**Comment by petersteneteg** [comment]
linuxdeployqt does not do any appImage it just copies dependencies... an appimage is a closer to a container https://appimage.org

**Comment by ResearchDaniel** [comment]
> linuxdeployqt does not do any appImage it just copies dependencies... an appimage is a closer to a container https://appimage.org

I think that you can "just" supply an -appimage argument to linuxdeployqt and then it should be more or less it
https://github.com/coryo/amphetype2/blob/2d41de3b0c19ab9286672ff0d6a7c11eadc13d9c/.travis/deploy.sh
From here:
https://github.com/AppImage/AppImageKit/wiki/Creating-AppImages/cc2441518975caca23e9ce2dba6f08a22c678d1e#4-run-linuxdeployqt-on-your-qt-application

**Pull request closed without merging**